core/src/main/clojure/xtdb/sql/plan.clj

```diff
@@ -1668,11 +1668,27 @@
           data-type (-> (.dataType ctx) (.accept (->CastArgsVisitor env)))]
       (handle-cast-expr ve data-type)))
 
+  (visitCollateExpr [_ ctx]
+    ;; TODO - apply collation
```

Comment:
What does this mean? We are now parsing `COLLATE` but are not really applying the logic?

Reply:
yeah - no idea what to do. in my understanding collation affects eg string ordering - but have no idea how and where to apply that. also - when trying to apply it - it might have to spread thru the system?

Reply:
If a query is using `COLLATE` then it's likely expecting results in some order. I am not saying we need to fully implement it, but it would be good to understand why a query is using `COLLATE` and then decide if or if not to honor the meaning.

Reply:
the query in this case is:
```sql
 SELECT c.oid,
  n.nspname,
  c.relname
FROM pg_catalog.pg_class c
     LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
WHERE c.relname OPERATOR(pg_catalog.~) '^(foo)$' COLLATE pg_catalog.default
  AND pg_catalog.pg_table_is_visible(c.oid)
ORDER BY 2, 3;
```
and default collation is used - so it would affect sort order - but given this is purely psql 'sugar' - think we can get away even in some cases when the tables are slightly 'misordered' (according to default collation)?
but then if users start using COLLATE - it might spell trouble. so maybe it should be documented that it's not supported?

Reply:
But `foo` is just an example here, right? This currently wouldn't work for `FOO` even though that is what the "real" pg would also work with. I am not saying you need to implement it, but lets' document it somewhere. Preferably some top-level comment on this PR and link to it from that `TODO`.

Reply:
If you want you can even open a new ticket for the feature `COLLATE` where you document what we currently do and what would need to happen to fully support `COLLATE`.

Reply:
opened #4020 to implement COLLATE